src/tracker_manager.cpp

```diff
@@ -278,8 +278,15 @@ constexpr tracker_request_flags_t tracker_request::i2p;
 
 #ifndef TORRENT_DISABLE_LOGGING
 		std::shared_ptr<request_callback> cb = c.lock();
-		if (cb) cb->debug_log("*** QUEUE_TRACKER_REQUEST [ listen_port: %d ]"
-			, req.listen_port);
+		if (cb) {
+			if (const auto announce_port = std::uint16_t(sett.get_int(settings_pack::announce_port))) {
+				cb->debug_log("*** QUEUE_TRACKER_REQUEST [ listen_port: %d, announce_port: %d ]"
+				, req.listen_port, announce_port);
+			} else {
+				cb->debug_log("*** QUEUE_TRACKER_REQUEST [ listen_port: %d ]"
+				, req.listen_port);
+			}
+		}
```

Comment:
I'm not sure this change is necessary.

Reply:
As in the entire change is unnecessary or you would rather have the a single `debug_log` call where `announce_port` would be `0` if not set?

Reply:
I don't think it's important (nor necessary) to preserve the distinction between "listen_port" and "announce_port". We only use this for tracker announcements anyway.

Reply:
Thanks! The change will be dropped anyway as `req.listen_port` will be overwritten in https://github.com/arvidn/libtorrent/blob/RC_2_0/src/session_impl.cpp#L1306-L1319

Waiting on the tests to pass to submit the change.